go-controller/pkg/ovn/pods.go

```diff
@@ -141,38 +141,31 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) (err error) {
 	}
 
 	logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
-	portInfo, err := oc.logicalPortCache.get(logicalPort)
-	if err != nil {
-		klog.Errorf(err.Error())
+	// using retryPods instead of logicalPortCache ensures we are deleting the right port
+	portInfo := oc.retryPods[logicalPort].needsDel
```

Comment:
this isn't thread safe. Need to use a helper function from the library here to return a copy. Or alternatively pass it into this function as an arg.

Reply:
oh nice catch!
damn this was a bad oversight on my part.

Reply:
fixed it.